profiler/profile.go

```diff
@@ -214,22 +214,25 @@ func (p *profiler) runProfile(pt ProfileType) ([]*profile, error) {
 	if err != nil {
 		return nil, err
 	}
-	profs := []*profile{{
-		name: t.Filename,
-		data: data,
-	}}
 	// Compute the deltaProf (will be nil if not enabled for this profile type).
 	deltaStart := time.Now()
 	deltaProf, err := p.deltaProfile(t, data)
 	if err != nil {
 		return nil, fmt.Errorf("delta profile error: %s", err)
 	}
-	// Report metrics and append deltaProf if not nil.
 	end := now()
 	tags := append(p.cfg.tags, pt.Tag())
-	// TODO(fg) stop uploading non-delta profiles in the next version of
-	// dd-trace-go after delta profiles are released.
-	if deltaProf != nil {
+	var profs []*profile
+	if !p.cfg.deltaProfiles || deltaProf == nil {
```

Comment:
NIT: I think the `!p.cfg.deltaProfiles` predicate is redundant here?

Reply:
You're right, good catch. I should have read the doc comment for `deltaProfile` :)

Reply:
No worries. Sometimes a redundant predicate can help to clarify intend, so I'm not entirely against it, just wanted to verify if that was the goal here. Thanks for the change! : )